google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessageTest.java

```diff
@@ -1019,6 +1019,76 @@ public void testRequired() throws Exception {
     }
   }
 
+  @Test
+  public void testRange() throws Exception {
+    TableSchema tableSchema =
+        TableSchema.newBuilder()
+            .addFields(
+                TableFieldSchema.newBuilder()
+                    .setName("range_date")
+                    .setType(TableFieldSchema.Type.RANGE)
+                    .setRangeElementType(
+                        TableFieldSchema.FieldElementType.newBuilder()
+                            .setType(TableFieldSchema.Type.DATE)
+                            .build())
+                    .setMode(TableFieldSchema.Mode.NULLABLE)
+                    .build())
+            .addFields(
+                TableFieldSchema.newBuilder()
+                    .setName("range_datetime")
+                    .setType(TableFieldSchema.Type.RANGE)
+                    .setRangeElementType(
+                        TableFieldSchema.FieldElementType.newBuilder()
+                            .setType(TableFieldSchema.Type.DATETIME)
+                            .build())
+                    .setMode(TableFieldSchema.Mode.NULLABLE)
+                    .build())
+            .addFields(
+                TableFieldSchema.newBuilder()
+                    .setName("range_timestamp")
+                    .setType(TableFieldSchema.Type.RANGE)
+                    .setRangeElementType(
+                        TableFieldSchema.FieldElementType.newBuilder()
+                            .setType(TableFieldSchema.Type.TIMESTAMP)
+                            .build())
+                    .setMode(TableFieldSchema.Mode.NULLABLE)
+                    .build())
+            .build();
+
+    TestRange expectedProto =
+        TestRange.newBuilder()
+            .setRangeDate(TestRangeDate.newBuilder().setStart(18262).setEnd(18627))
+            .setRangeDatetime(
+                TestRangeDatetime.newBuilder().setStart(1715360343).setEnd(1715446743))
+            .setRangeTimestamp(
+                TestRangeTimestamp.newBuilder().setStart(1715360343).setEnd(1715446743))
+            .build();
+
+    JSONArray data = new JSONArray();
+    JSONObject row = new JSONObject();
+
+    JSONObject rangeDate = new JSONObject();
+    rangeDate.put("start", 18262);
+    rangeDate.put("end", 18627);
+    row.put("range_date", rangeDate);
```

Comment:
Do we require cases to match here? 
We should at least test field name with different case? 

Can "start" and "end" be case insensitive? This is more like an open question.